### DISCUSSION TITLE
警告: 潜在的なセキュリティリスクあり

### DISCUSSION THREAD

**Discussion by scivola** [increments/qiita-discussions#418] [ヘルプ]
Firefox で

```
警告: 潜在的なセキュリティリスクあり
```

として接続できないことがあります。

```
エラーコード: SSL_ERROR_BAD_CERT_DOMAIN
```

です。

再現条件はよくわかりません。

**Comment by tanaka-asami415** [comment]
@scivola 
https://github.com/increments/qiita-discussions/discussions/374 の方に返信させていただきましたので、ご確認いただけますと幸いです🙇

**Comment by scivola** [comment]
情報を追加します。
エラーコードの前に以下のようなメッセージが表示されます。

ほとんど同じ文面で，ドメイン名はいくつか種類があるようです。

```
ウェブサイトは証明書で同一性を証明します。qiita.com は無効な証明書を使用しているため、Firefox はこのサイトを信頼しません。 この証明書は cell1.nrt.gamma.dashboard.capstone.ai.aws.dev にだけ有効なものです。
```

```
ウェブサイトは証明書で同一性を証明します。qiita.com は無効な証明書を使用しているため、Firefox はこのサイトを信頼しません。この証明書は次のドメイン名にのみ有効です: *.company.works-hi.com, company.works-hi.com
```

```
ウェブサイトは証明書で同一性を証明します。qiita.com は無効な証明書を使用しているため、Firefox はこのサイトを信頼しません。この証明書は次のドメイン名にのみ有効です: blossompj.dev, *.blossompj.dev
```

Firefox 109.0 および Firefox Developper Edition 110.0b3 で現象を確認しています。

**Reply by yoshi389111** [reply]
通りすがりの者です。

掲示のドメインを見る限り、あまり Qiita の物っぽくない気がするので、広告など別オリジンの接続の証明書で不正な(SANが正しくない)ものが使われているのではないでしょうか？

例えば、広告ブロッカーとか入れたら解決したりしないかな？（恒久的な対策というよりは、事象の分析のため、という意味が強いですが）

追記：firefoxは 102 ぐらいで、SANに値がない場合のフォールバックとして CN を見る挙動を削除したという話を[見かけ](https://www.clear-code.com/blog/2022/12/14/firefox-esr102-no-san-fallback-error.html)ました。[RFC9110の修正](https://eng-blog.iij.ad.jp/archives/14820)に準拠したのかな？　このあたりが影響しているんですかね？

**Reply by scivola** [reply]
ありがとうございます。
私も，メッセージに書かれたドメインの少なくとも一つは qiita のものではなく，qiita の広告主さんのものらしいので，広告の問題では？とは（後になって）思いました。

広告ブロッカーは，AdBlocker Ultimate が有効な状態と無効な状態の両方で現象が再現することを確認しました。

**Comment by tanaka-asami415** [comment]
https://github.com/increments/qiita-discussions/discussions/374#discussioncomment-4803790 にて解決済み。
本Discussionはクローズさせていただきます。
